tests/SpecTests/SearchIndexSpecTest.php

```diff
@@ -0,0 +1,190 @@
+
+        foreach ($names as $key => $name) {
+            $index = $indexes[$key];
+            $this->assertSame($name, $index->name);
+
+            // Convert to JSON to compare nested associative arrays and nested objects
+            $this->assertJsonStringEqualsJsonString(
+                json_encode($mapping, JSON_THROW_ON_ERROR),
+                json_encode($index->latestDefinition, JSON_THROW_ON_ERROR),
+            );
+        }
+    }
+
+    public function testDropSearchIndexes(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name],
+        );
+        $this->assertSame($name, $createdName);
+
+        $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $collection->dropSearchIndex($name);
+
+        $indexes = $this->waitForIndexes($collection, fn (array $indexes): bool => count($indexes) === 0);
+
+        $this->assertCount(0, $indexes);
+    }
+
+    public function testUpdateSearchIndex(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name],
+        );
+        $this->assertSame($name, $createdName);
+
+        $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $mapping = ['mappings' => ['dynamic' => true]];
+        $collection->updateSearchIndex($name, $mapping);
+
+        [$index] = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $this->assertSame($name, $index->name);
+
+        // Convert to JSON to compare nested associative arrays and nested objects
+        $this->assertJsonStringEqualsJsonString(
+            json_encode($mapping, JSON_THROW_ON_ERROR),
+            json_encode($index->latestDefinition, JSON_THROW_ON_ERROR),
+        );
+    }
+
+    public function testDropSearchIndexSuppressNamespaceNotFoundError(): void
+    {
+        $collection = $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
+
+        $collection->dropSearchIndex('test-seach-index');
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * Randomize the collection name to avoid duplicate index names when running tests concurrently.
+     * Search index operations are asynchronous and can take up to a few minutes.
+     */
+    protected function getCollectionName(): string
+    {
+        return sprintf('%s.%s', parent::getCollectionName(), bin2hex(random_bytes(5)));
```

Comment:
I assume this may address my previous concern with randomized index names. Feel free to resolve both of these comments if so.

Reply:
Yes, that randomize the collection name for the whole class.